ross/fluid_flow/tilting_pad.py

```diff
@@ -1119,26 +1125,26 @@ def _forces(self):
                         fdiss = 2 * (
                             (
                                 self.Cr * hP * (up_a - uw_a) / dksi
-                                - n_p * self.Cr * dhdksi_p * (up_a - us_a) / self.dN
+                                - np2 * self.Cr * dhdksi_p * (up_a - us_a) / self.dN
                             )
                             ** 2
                             + (self.betha_s * self.Rs * (vn_a - vp_a) / self.dN) ** 2
                             + (
                                 self.betha_s * self.Rs * (up_a - us_a) / self.dN
                                 + self.Cr * hP * (vp_a - vw_a) / dksi
-                                - n_p * self.Cr * dhdksi_p * (vp_a - vs_a) / dksi
+                                - np2 * self.Cr * dhdksi_p * (vp_a - vs_a) / dksi
                             )
                             ** 2
                             + (
                                 self.Cr * hP * (wp_a - ww_a) / dksi
-                                - n_p * self.Cr * dhdksi_p * (wp_a - ws_a) / self.dN
+                                - np2 * self.Cr * dhdksi_p * (wp_a - ws_a) / self.dN
                             )
                             ** 2
                             + (self.betha_s * self.Rs * (wp_a - ws_a) / dksi) ** 2
                         )
 
                         # Source term ----------------
-                        Bp = (
+                        Bp = Decimal(
```

Comment:
Hi there @verg1lio @marcusfilipesr !
Just a question, why did you guys choose to use Decimal in this case? Floats are not ok here?

Reply:
Hi! We believe that the machine precision might be a problem, since our 'b' array is almost matching the matlab one, for the exception of a few values where Matlab has 1e-12 or smaller and Python has 0. To test this we used Decimal to increased precision, but it may not be the root cause of the issue.

Reply:
By using the pseudo inverse instead of standard linalg.solve this problem may be solved. We will work on ironing out this now.

Reply:
Hi @raphaeltimbo !

We managed to find a reliable and robust solution to the problem of the linear system. As of now, all the main code is running as intended and matching our Matlab original program. Any discrepancies measured at the moment are of machine precision. We (@verg1lio and @marcusfilipesr ) will be moving into the post-processing phase, leading to the K and C coefficients from now on.

Cheers, and have a good evening!